hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/fetcher/LabelFetcher.java

```diff
@@ -16,19 +16,20 @@ public LabelFetcher() {
   }
 
   static final class LabelFetcherImpl implements DataFetcher<CompletableFuture<LabelResultSet>> {
-    private final ContextualRequestBuilder requestBuilder;
+    private final LabelJoinerBuilder requestBuilder;
     private final LabelDao labelDao;
 
     @Inject
-    LabelFetcherImpl(ContextualRequestBuilder requestBuilder, LabelDao labelDao) {
+    LabelFetcherImpl(LabelJoinerBuilder requestBuilder, LabelDao labelDao) {
```

Comment:
I think we're combing two different things here. The label joiner is meant to join labels to other objects with label references, here we've combined the other direction - that is, joining other objects onto labels. I think it likely makes sense to separate them, not just for standard separation of concerns (certainly this reason too), but also to prevent infinitely complex queries. If I try to go either entity -> label -> entity or label -> entity -> label, we want it to either fail (ideal) or at a minimum skip the recursion.

Reply:
Extracted entity joiner to a different class